mobile-browser-based-version/src/helpers/training/training_manager.js

```diff
@@ -144,79 +150,263 @@ export class TrainingManager {
   async _trainLocally() {
     const info = this.task.trainingInformation;
 
-    await this.model.fit(this.data, this.labels, {
-      initialEpoch: this.model.getUserDefinedMetadata().epoch,
-      epochs: info.epochs ?? MANY_EPOCHS,
-      batchSize: info.batchSize,
-      validationSplit: info.validationSplit,
-      shuffle: true,
-      callbacks: {
-        onEpochEnd: async (epoch, logs) => {
-          this.trainingInformant.updateGraph(
-            epoch + 1,
-            (logs.val_acc * 100).toFixed(2),
-            (logs.acc * 100).toFixed(2)
-          );
-          console.log(
-            `EPOCH (${epoch + 1}):
+    var dataSize = -1;
+
+    // If info resize is not defined it will default to Fast Training
```

Comment:
what is the semantics of 'resize'? this seems to be in a code part which is general, not image-specific, so a bit surprising to see the 8*3*... hardcoded below

Reply:
The only problem I see is that in general preporcessing for tabular datasets will not make the size of the dataset much bigger so defaulting those the training mode that is faster is a satisfactory solution?

Reply:
So in that case calculating the data size only for images should be fine?

Reply:
not sure what 'resize' means here. do you mean batch size? also the overall dataset size (memory used) is rougly the same anyway, no matter if batching is turned on or off? (or do you mean to store the results of the preprocessing, or compute on the fly?). please clarify in the code comments once you have time. not urgent of course